src/renderer/components/cluster-settings/components/cluster-node-shell-setting.tsx

```diff
@@ -0,0 +1,140 @@
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import type { Cluster } from "../../../../main/cluster";
+import { autorun, makeObservable, observable } from "mobx";
+import { SubTitle } from "../../layout/sub-title";
+import React from "react";
+import { Input } from "../../input/input";
+import { disposeOnUnmount, observer } from "mobx-react";
+import { Icon } from "../../icon/icon";
+import { initialNodeShellImage } from "../../../../common/cluster-store";
+import { FormSwitch, Switcher } from "../../switch";
+
+interface Props {
+  cluster: Cluster;
+}
+
+@observer
+export class ClusterNodeShellSetting extends React.Component<Props> {
+  @observable nodeShellImage = "";
+  @observable imagePullSecret = "";
+  @observable usePrivateRegistry = false;
+
+  constructor(props: Props) {
+    super(props);
+    makeObservable(this);
+  }
+
+  componentDidMount() {
+    this.usePrivateRegistry = !!this.props.cluster.getImagePullSecret();
+
+    disposeOnUnmount(this,
+      autorun(() => {
+        this.nodeShellImage = this.props.cluster.getNodeShellImage();
+        this.imagePullSecret = this.props.cluster.getImagePullSecret();
+      })
+    );
+  }
+
+  onImageChange = (value: string) => {
+    this.nodeShellImage = value;
+  };
+
+  onSecretChange = (value: string) => {
+    this.imagePullSecret = value;
+  };
+
+  saveImage = () => {
+    this.props.cluster.preferences.nodeShellImage = this.nodeShellImage;
+  };
+
+  saveSecret = () => {
+    this.props.cluster.preferences.imagePullSecret = this.imagePullSecret;
+  };
+
+  resetImage = () => {
+    this.nodeShellImage = initialNodeShellImage; //revert to default
+  };
+
+  clearSecret = () => {
+    this.imagePullSecret = "";
+  };
+
+  onSwitchChange = () => {
+    this.usePrivateRegistry = !this.usePrivateRegistry;
+
+    if (!this.usePrivateRegistry) {
+      this.imagePullSecret = "";
+      this.saveSecret();
+    }
+  };
+
+  render() {
+    const resetImageIcon = <Icon small material="close" onClick={this.resetImage} tooltip={"Reset"}/>;
+    const clearSecretIcon = <Icon small material="close" onClick={this.clearSecret} tooltip={"Clear"}/>;
+
+    return (
+      <>
+        <section>
+          <SubTitle title="Node shell image" id="node-shell-image"/>
+          <Input
+            theme="round-black"
+            value={this.nodeShellImage}
+            onChange={this.onImageChange}
+            onBlur={this.saveImage}
+            iconRight={resetImageIcon}
+          />
+          <small className="hint">
+            Node shell image. Used for creating node shell pod.
+          </small>
+        </section>
+        <section>
+          <FormSwitch
+            control={
+              <Switcher
+                checked={this.usePrivateRegistry}
+                onClick={this.onSwitchChange}
+                name="private-registry"
+              />
+            }
+            label="Use private registry"
+          />
```

Comment:
I think this will be confusing for users because it is essentially a "clear imagePullSecret" button. I think it would be better to not have it and just add something like "Optional" to the hint for the `imagePullSecret`